flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -475,4 +476,23 @@ public void setRootConfiguration(ReadableConfig rootConfiguration) {
     public static TableConfig getDefault() {
         return new TableConfig();
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Actually using a mutable object as a map key is not a good practice. I'm wondering if there is another way to implement the plan cache, such as invalidating the cache when config is changed?